packages/cursorless-vscode/src/ide/vscode/VscodeCommandHistory.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  CommandComplete,
+  CommandHistory,
+  CommandHistoryItem,
+  Disposable,
+  FileSystem,
+} from "@cursorless/common";
+import * as fs from "fs/promises";
+import { mkdirSync } from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { VscodeIDE } from "./VscodeIDE";
+
+const dirName = "commandHistory";
+const filePrefix = "cursorlessCommandHistory";
+const settingSection = "cursorless";
+const settingName = "commandHistory";
+const settingFullName = `${settingSection}.${settingName}`;
+
+export class VscodeCommandHistory implements CommandHistory {
+  private readonly dirPath: string;
+  private readonly cursorlessVersion: string;
+  private disposable: Disposable;
+  private active: boolean = false;
+
+  constructor(ide: VscodeIDE, fileSystem: FileSystem) {
+    this.evaluateSetting = this.evaluateSetting.bind(this);
+    this.cursorlessVersion = ide.cursorlessVersion;
+    this.dirPath = path.join(fileSystem.cursorlessDir, dirName);
+
+    mkdirSync(this.dirPath, { recursive: true });
```

Comment:
I would move the mkdir to the append method for these reasons:
- Ensures the directory is created right before writing to it. For example with the current mechanism if I delete the directory mid session, the append will throw I think
- No need to wait for directory to be created - currently blocking activate path

Reply:
The director is owned by Cursorless and the uses should not remove it.